core/src/actors/mining_manager/handlers.rs

```diff
@@ -0,0 +1,100 @@
+// ACTOR MESSAGE HANDLERS
+////////////////////////////////////////////////////////////////////////////////////////
+
+/// Payload for the notification for all epochs
+#[derive(Clone, Debug)]
+pub struct EveryEpochPayload;
+
+/// Handler for EpochNotification<EveryEpochPayload>
+impl Handler<EpochNotification<EveryEpochPayload>> for MiningManager {
+    type Result = ();
+
+    fn handle(&mut self, msg: EpochNotification<EveryEpochPayload>, ctx: &mut Context<Self>) {
+        debug!("Periodic epoch notification received {:?}", msg.checkpoint);
+
+        let chain_manager_addr = System::current().registry().get::<ChainManager>();
+        chain_manager_addr
+            .send(GetHighestCheckpointBeacon)
+            .into_actor(self)
+            .then(move |beacon_msg, act, ctx| {
+                // Check eligibility
+                // S(H(beacon))
+                let mut beacon = match beacon_msg {
+                    Ok(Ok(b)) => b,
+                    _ => return actix::fut::err(()),
+                };
+                if beacon.checkpoint > msg.checkpoint {
+                    // We got a block from the future
+                    error!("The current highest checkpoint beacon is from the future");
+                    return actix::fut::err(());
+                }
+                if beacon.checkpoint == msg.checkpoint {
+                    // For some reason we already got a valid block for this epoch
+                    // TODO: Check eligibility anyway?
+                }
+                // TODO: this is correct, right?
+                beacon.checkpoint = msg.checkpoint;
+                let beacon_hash = beacon.hash();
+                let private_key = 1;
+
+                // TODO: send Sign message to CryptoManager
+                let sign = |x, _k| match x {
+                    Hash::SHA256(x) => x,
+                };
+                let signed_beacon_hash = sign(beacon_hash, private_key);
+                // Currently, every hash is valid
+                // Fake signature which will be accepted anyway
+                let signature = Signature::Secp256k1(Secp256k1Signature {
+                    r: signed_beacon_hash,
+                    s: signed_beacon_hash,
+                    v: 0,
+                });
+                let leadership_proof = LeadershipProof {
+                    block_sig: Some(signature),
+                    influence: 0,
+                };
+
+                // Send ValidatePoE message to ReputationManager
+                let reputation_manager_addr =
+                    System::current().registry().get::<ReputationManager>();
+                reputation_manager_addr
+                    .send(ValidatePoE {
+                        beacon,
+                        proof: leadership_proof,
+                    })
+                    .into_actor(act)
+                    .drop_err()
+                    .and_then(move |eligible, _act, _ctx| {
+                        if eligible {
```

Comment:
Let's log this as `info`.

Reply:
`info!("Mined a block!");` would be enough or do we want to log the proof of eligibility?

Reply:
Something similar to this should do the job:
```rust
info!("Discovered eligibility for mining a block for epoch #{:?}");
```